include/circt/Dialect/HW/HWStructure.td

```diff
@@ -231,6 +231,10 @@ def HWModuleOp : HWModuleOpBase<"module",
 
     /// Verifies the body of the function.
     LogicalResult verifyBody();
+
+    void appendPorts(
+                  ArrayRef<std::tuple<StringAttr, Type, Location>> insertInputs,
+                  ArrayRef<std::pair<StringAttr, Value>> insertOutputs);
```

Comment:
What's difference from `modifyPorts` which we are going to deprecate?

Reply:
Good point, thanks for bringing it up, I will add comments here. 
Few reasons,
1. Stop using the deprecated methods.
2. Implement a simpler API, that only appends the input and then output ports, the most-common/only use case. Avoids the generic methods, that can insert/delete at random port indices. 
3. Avoid any performance foot-guns, the deprecated methods can have performance issues, implementing the fresh API that we discuss and properly inspect to ensure there are no obvious performance issues.
4. At least rhodium can stop using the deprecated API, and switch to this, if its approved.
I will commit this separately, so if it makes sense to discuss this more broadly, I will create a new PR with this change.
What do you think ?

Reply:
Sorry for delayed replay. TBH I don't find much difference between deprecated functions. The intention for the deprecation was "it doesn't make sense to have an API which takes input and output ports separately" (https://github.com/llvm/circt/blob/5d8cf69cf222ec8776b1680d4d1de7e0e2aa4a86/lib/Dialect/HW/HWOps.cpp#L675-L679) so I'm not sure it's a right direction.

Reply:
Thanks for your suggestions Hideto, I updated the interface, we need the output Values to update the `OutputOp` also added the input values, that can be set as the block arguments are added. This avoids the need to know the argument index of the input args.